### PR TITLE
Align YAML with the Helm Chart PR

### PR DESCRIPTION
**Pull request by dvonthenen**
**What this PR does / why we need it**:
This aligns the example/provided YAML with that of the recent Helm Chart YAML PR: https://github.com/helm/charts/pull/23523

Ideally, this should be merged together in order to prevent YAML configuration drift.

**Which issue this PR fixes**:
NA

**Special notes for your reviewer**:
Tested during Chart PR

**Release note**:
NA

### CODE REVIEW

**Comment by dvonthenen** [comment]
Most of this is just moving stuff around... the chart had things in a certain order and the YAML had it in a different one. This is mostly just to make doing the diff between the two easier to facilitate updates to the chart quicker.

**Comment by dvonthenen** [comment]
Tested on 1.18.12 and works great!

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm
/hold

(unhold whenever)

File: releases/v1.18/vsphere-cloud-controller-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - "coordination.k8s.io"
+    resources:
+      - leases
+    verbs:
+      - get
```
Comment (andrewsykim) [inline releases/v1.18/vsphere-cloud-controller-manager.yaml:191]: I think we want this to include create and update going forward?

Comment (dvonthenen) [reply releases/v1.18/vsphere-cloud-controller-manager.yaml:192]: yup

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm
/hold cancel

**Pull request merged**